org.flatpak.popout3d.yml

```diff
@@ -0,0 +1,204 @@
+app-id: com.github.popout3-apps.popout3d
+
+runtime: org.gnome.Platform
+# 3.30 is deprecated, but the hugin stuff won't work with later versions.
+# There is now a Flatpak for hugin which might not have been available when 
+# this manifest was written. There may be something useful there.
+
+runtime-version: "3.30"
+sdk: org.gnome.Sdk
+
+command: popout3d
+rename-desktop-file: popout3d.desktop
+rename-icon: popout3d_logo
+copy-icon: true
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --filesystem=xdg-run/dconf
+  - --filesystem=~/.config/dconf:ro
+  - --talk-name=ca.desrt.dconf
+  - --env=DCONF_USER_CONFIG_DIR=.config/dconf
+  - --filesystem=host:rw
```

Comment:
You should use GtkFileChooserNative to open and save files.

Reply:
Is that mandatory or a recommendation? It will take me a while to learn itt, so I'd prefer to get this sorted first and do that in the next version.

Reply:
> filechooserdialog = Gtk.**FileChooserNative**(title="Select a file", action=Gtk.FileChooserAction.OPEN)
![b](https://user-images.githubusercontent.com/20310498/104966132-13819700-59d8-11eb-8967-6cf7c9984802.png)
opens a completely new window with an extra icon to the list of open applications - very clumsy and confusing to use. It doesn't allow adding a reference to a parent window, so I tried
> filechooserdialog = Gtk.**FileChooserNative.new**('Select a file', **self.window**, Gtk.FileChooserAction.OPEN,'OK','Cancel')
('window' is the name of the application's top-level window)
no better, it still opens a completely separate window. 

Rather than use something that looks absolutely awful on any system including the native system, I have gone back to the original which looks great on its native system but might be less than perfectly matched on another system. 

Unless you have a suggestion for a better way to do this?

Reply:
> filechooserdialog = Gtk.**FileChooserNative**(title="Select a file", action=Gtk.FileChooserAction.OPEN)

![b](https://user-images.githubusercontent.com/20310498/104966132-13819700-59d8-11eb-8967-6cf7c9984802.png)

opens a completely new window with an extra icon to the list of open applications - very clumsy and confusing to use. It doesn't allow adding a reference to a parent window, so I tried
> filechooserdialog = Gtk.**FileChooserNative.new**('Select a file', **self.window**, Gtk.FileChooserAction.OPEN,'OK','Cancel')

'window' is the name of the application's top-level window. This is no better, it still opens a completely separate window. 

Rather than use something that looks absolutely awful on any system including the native system, I will go back to the original which looks great on its native system but might be less than perfectly matched on another system. 

Unless you have a suggestion for a better way to do this?